solidity/contracts/relays/OpenRelay.sol

```diff
@@ -17,25 +17,56 @@ contract OpenRelay is IOpenRelay {
   /// @inheritdoc IOpenRelay
   uint32 public constant BASE = 10_000;
 
+  mapping(IAutomationVault _automationVault => mapping(address _job => PaymentData _paymentData)) public paymentsData;
+
+  function setExtraPayment(IAutomationVault _automationVault, address _job, PaymentData memory _paymentData) external {
+    if (_automationVault.owner() != msg.sender) revert OpenRelay_NotVaultOwner();
+
+    paymentsData[_automationVault][_job] = _paymentData;
+
+    emit ExtraPaymentSetted(_automationVault, _job, _paymentData);
+  }
+
   /// @inheritdoc IOpenRelay
   function exec(
     IAutomationVault _automationVault,
     IAutomationVault.ExecData[] calldata _execData,
     address _feeRecipient
   ) external {
     if (_execData.length == 0) revert OpenRelay_NoExecData();
+    IAutomationVault.ExecData[] memory _newExecData = new IAutomationVault.ExecData[](1);
+    IAutomationVault.FeeData[] memory _feeData = new IAutomationVault.FeeData[](_execData.length);
+    PaymentData memory _paymentData;
+
+    uint256 _initialGas;
+    uint256 _gasSpent;
+    uint256 _payment;
+
+    for (uint256 _i; _i < _execData.length;) {
+      _paymentData = paymentsData[_automationVault][_execData[_i].job];
+      _newExecData[0] = _execData[_i];
```

Comment:
is it cheaper this way? 👀

Reply:
ahhh it must be an array, why?

Reply:
The whole xkeeper system was designed by passing an array to work from jobs and selectors to encourage calling more than one, if we change it, we would have to change the whole logic of all the relays and vaults. I am not against it but we would have to agree on it.